tests/filecheck/transforms/snax-pin-core-end-to-end.mlir

```diff
@@ -0,0 +1,84 @@
+// RUN: ./compiler/snax-opt %s -p dispatch-regions{nb_cores=3\ pin_to_constants=true},function-constant-pinning,snax-to-func | mlir-opt-17 --canonicalize --inline| filecheck %s 
```

Comment:
I guess I should remove the --inline, since @AntonLydike specifically outlined the function calls because I requested to do so :( . @AntonLydike can we make the outlining optional at some point do you think?